cpp/tests/community/new_leiden_test.cpp

```diff
@@ -0,0 +1,410 @@
+    }
+
+    auto [graph, edge_weights, d_renumber_map_labels] =
+      cugraph::test::construct_graph<vertex_t, edge_t, weight_t, false, false>(
+        handle, input_usecase, true, renumber);
+
+    if (cugraph::test::g_perf) {
+      RAFT_CUDA_TRY(cudaDeviceSynchronize());  // for consistent performance measurement
+      hr_timer.stop();
+      hr_timer.display_and_clear(std::cout);
+    }
+
+    auto graph_view = graph.view();
+    auto edge_weight_view =
+      edge_weights ? std::make_optional((*edge_weights).view()) : std::nullopt;
+
+    // "FIXME": remove this check once we drop support for Pascal
+    //
+    // Calling leiden on Pascal will throw an exception, we'll check that
+    // this is the behavior while we still support Pascal (device_prop.major < 7)
+    //
+    cudaDeviceProp device_prop;
+    RAFT_CUDA_TRY(cudaGetDeviceProperties(&device_prop, 0));
+
+    if (cugraph::test::g_perf) {
+      RAFT_CUDA_TRY(cudaDeviceSynchronize());  // for consistent performance measurement
+      hr_timer.start("Leiden");
+    }
+
+    if (device_prop.major < 7) {
+      EXPECT_THROW(leiden(graph_view,
+                          edge_weight_view,
+                          graph_view.local_vertex_partition_range_size(),
+                          leiden_usecase.max_level_,
+                          leiden_usecase.resolution_,
+                          leiden_usecase.check_correctness_,
+                          leiden_usecase.expected_level_,
+                          leiden_usecase.expected_modularity_),
+                   cugraph::logic_error);
+    } else {
+      leiden(graph_view,
+             edge_weight_view,
+             graph_view.local_vertex_partition_range_size(),
+             leiden_usecase.max_level_,
+             leiden_usecase.resolution_,
+             leiden_usecase.check_correctness_,
+             leiden_usecase.expected_level_,
+             leiden_usecase.expected_modularity_);
+    }
+
+    if (cugraph::test::g_perf) {
+      RAFT_CUDA_TRY(cudaDeviceSynchronize());  // for consistent performance measurement
+      hr_timer.stop();
+      hr_timer.display_and_clear(std::cout);
+    }
+  }
+
+  template <typename vertex_t, typename edge_t, typename weight_t>
+  void leiden(
+    cugraph::graph_view_t<vertex_t, edge_t, false, false> const& graph_view,
+    std::optional<cugraph::edge_property_view_t<edge_t, weight_t const*>> edge_weight_view,
+    vertex_t num_vertices,
+    size_t max_level,
+    float resolution,
+    bool check_correctness,
+    int expected_level,
+    float expected_modularity)
+  {
+    raft::handle_t handle{};
+
+    rmm::device_uvector<vertex_t> clustering_v(num_vertices, handle.get_stream());
+    size_t level;
+    weight_t modularity;
+
+    std::tie(level, modularity) = cugraph::leiden(
+      handle, graph_view, edge_weight_view, clustering_v.data(), max_level, resolution);
+
+    RAFT_CUDA_TRY(cudaDeviceSynchronize());  // for consistent performance measurement
```

Comment:
What is this? This shouldn't be necessary for consistent performance measurement. And if this function is returning (level, modularity), these host variables should be valid right after this function returns.